BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoPUEO_Symmetric.py

```diff
@@ -0,0 +1,133 @@
+    1016.70e6, 1033.40e6, 1050.00e6, 1066.70e6
+    ]
+numFreq = 60
+'''
+
+freqList = [f'{val}' for val in freqVals] # This is in the format that pueoSim wants
+big_freq_list = freqList * 6 # This is the same thing, just 6 times :) (this is to get the gain at different angles at each frequency, we have 6 angles)
+
+# Adds arguments when running code
+parser = argparse.ArgumentParser()
+parser.add_argument("NPOP", help="How many antennas are being simulated each generation?", type=int)
+parser.add_argument("WorkingDir", help="Evolutionary_Loop directory", type=Path)
+parser.add_argument("RunName", help="Run Name directory where everything will be stored", type=str)
+parser.add_argument("gen", help="The generation the loop is on", type=int)
+parser.add_argument("indiv", help="The individual in the population", type=int)
+g = parser.parse_args()
+
+# Function to read in files
+def readFile(indiv, freqNum, col):
+    uanname = Path(f'{g.WorkingDir}/Run_Outputs/{g.RunName}/{g.gen}_{indiv}_{freqNum}.uan')
+    #uanname = Path(f'{g.WorkingDir}/{g.RunName}/uan_files/{g.gen}_uan_files/{indiv}/{g.gen}_{indiv}_{freqNum}.uan')
+    #pulling data from the uan file                                                                                         
+    data = np.genfromtxt(uanname, unpack=True, skip_header=18).tolist()
+
+    return (data[col]) #returns Gain(Theta) column THIS WILL NEED TO BE CHANGED FOR WHEN WE HAVE WORKING XF OUTPUTS (maybe make column input variable for function?)                                                                               
+## Make a function to create a list for each of the columns (V and H gains)
+## This function reads in the gain at each phi for the desired thetas
+def getGains(indiv):
+	vpolGains = []
+	hpolGains = []
+
+	for freq in range(numFreq):
+		vpolGains.append(readFile(indiv, freq+1, 2))
+		hpolGains.append(readFile(indiv, freq+1, 3))
+
+	return vpolGains, hpolGains
+
+## This function writes the data to a file
+def writeGains(data, freqList, fileName):
+	with open(fileName, "a") as f:
+		for i in range(numFreq):
+			f.write(str(freqList[i]) + " " + str(data[i]) + "\n")
+	return 0
+
+
+## Here's the path to output to
+fileHeader = g.WorkingDir / 'Test_Outputs' 
+## Loop over the number of individuals
+for indiv in range(1, g.NPOP+1):
+		## Let's call the functions
+		## Start by getting the gain data
+		vpolData, hpolData = getGains(indiv)
+		## We're treating vv and hh as the same and vh and hv as the same
+		## So for the _0 files, it's pretty simple:
+		writeGains(np.transpose(vpolData).tolist()[0], freqList, fileHeader / f'vv_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(vpolData).tolist()[0], freqList, fileHeader / f'hh_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(hpolData).tolist()[0], freqList, fileHeader / f'vh_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(hpolData).tolist()[0], freqList, fileHeader / f'hv_0_{g.gen}_{indiv}')
```

Comment:
The `transpose` part for `vpolData` and `hpolData` is duplicated.  Not sure how expensive it is, but you could avoid some duplication and make the code a bit more readable by using intermediate vars for `np.transpose(...).tolist()[0]`.  An even better option would be to use them for just `np.transpose(...).tolist()` so that you can reuse it in the loops below too, avoiding even more unnecessary calls to `transpose`/`tolist`.

Reply:
Good point! I'll make a new commit for this and a couple other notes that I think will be easier for me from the terminal than here on github.